### PR TITLE
Switch to composite USB gadget, enable terminal on ttyGS1

### PR DESCRIPTION
**Pull request by jacobmcnamee**
- Set up 2 x ACM (serial) + ECM (ethernet) USB gadget.
- Enable terminal on ttyGS1

/cc @swift-nav/firmware

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v0.1-18-g853f42e
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v0.1-18-g853f42e/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v0.1-18-g853f42e/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v0.1-18-g853f42e

File: board/piksiv3/rootfs-overlay/etc/init.d/S38usb_gadget [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+name="usb_gadget"
+
+VENDOR_ID="0x0000"
+PRODUCT_ID="0x0000"
```
Comment (fnoble) [inline board/piksiv3/rootfs-overlay/etc/init.d/S38usb_gadget]: Do we need a proper VID/PID here?

Comment (jacobmcnamee) [inline board/piksiv3/rootfs-overlay/etc/init.d/S38usb_gadget]: "Need" might be a strong word. It works fine on mac. I haven't tested anything else. I assume windows will complain.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-220-g393690f
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-220-g393690f/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-220-g393690f/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-220-g393690f

File: board/piksiv3/rootfs-overlay/etc/init.d/S38usb_gadget (hunk 1 above)
Comment (lbara) [inline board/piksiv3/rootfs-overlay/etc/init.d/S38usb_gadget]: I tried this as-is and the device was not usable on Windows (it was on Linux).

I changed the vendor ID and product ID and I was able to use it on Windows, Linux and MAC.

**Comment by lbara** [comment]
I would really love to merge this pull request.
Fergus was correct in saying that VID/PID of 0x0 were giving problems in Windows.
I tested this code with the different VID/PID and it correctly works on mac, linux and windows.

Caveat:
* on Windows, the "first" COM is not necessarily the SBP one (they are enumerated inconsistently)
So we cannot really deterministically say what port is what.

Dennis rightfully thinks that this is a pretty poor customer experience, but I propose we check this is right now because it will allow to un-brick piksi when the ports are all configured for output or the configuration is lost.

I ran this by Akshay and he was okay with this solution.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-develop-2017072120-11-g2bf92c9
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072120-11-g2bf92c9/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-develop-2017072120-11-g2bf92c9/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072120-11-g2bf92c9

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-develop-2017072120-14-ga0eb3d3
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072120-14-ga0eb3d3/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-develop-2017072120-14-ga0eb3d3/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072120-14-ga0eb3d3

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-develop-2017072120-15-geacf9ca
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072120-15-geacf9ca/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-develop-2017072120-15-geacf9ca/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072120-15-geacf9ca

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-develop-2017072819-14-gd5d3b0e
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072819-14-gd5d3b0e/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-develop-2017072819-14-gd5d3b0e/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-develop-2017072819-14-gd5d3b0e

**Pull request merged**